src/huggingface_hub/hf_api.py

```diff
@@ -1844,10 +1844,8 @@ def upload_file(
                 The git revision to commit from. Defaults to the head of the
                 `"main"` branch.
             identical_ok (`bool`, *optional*, defaults to `True`):
-                When set to false, will raise an [HTTPError](
-                https://2.python-requests.org/en/master/api/#requests.HTTPError)
-                when the file you're trying to upload already exists on the hub
-                and its content did not change.
+                Deprecated: will be removed in 0.10.0.
```

Comment:
```suggestion
                Deprecated: will be removed in 0.11.
```

we release rather frequently, it's nicer to give users a bit of time.